hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/cancelairdrops/TokenCancelAirdropDecoderTest.java

```diff
@@ -0,0 +1,327 @@
+    }
+
+    @Test
+    void failsIfPendingAirdropsAboveLimit() {
+        // given:
+        given(attempt.configuration()).willReturn(configuration);
+        given(configuration.getConfigData(TokensConfig.class)).willReturn(tokensConfig);
+        given(tokensConfig.maxAllowedPendingAirdropsToCancel()).willReturn(10);
+
+        final var encoded = Bytes.wrapByteBuffer(CANCEL_AIRDROP.encodeCall(Tuple.singleton(new Tuple[] {
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L)
+        })));
+        given(attempt.inputBytes()).willReturn(encoded.toArrayUnsafe());
+
+        assertThrows(HandleException.class, () -> subject.decodeCancelAirdrop(attempt));
```

Comment:
Is it valuable to check the code is `PENDING_AIRDROP_ID_LIST_TOO_LONG`, or unnecessary?

Reply:
Yes, added check